contracts/RewardPool.sol

```diff
@@ -153,6 +160,46 @@ contract RewardPool is Initializable, Versionable, Ownable {
     return true;
   }
 
+  function recoverTokens(
+    address rewardProgramID,
+    address token,
+    uint256 amount
+  ) external returns (bool) {
+    address rewardProgramAdmin = RewardManager(rewardManager)
+      .rewardProgramAdmins(rewardProgramID);
+    require(rewardProgramAdmin != ZERO_ADDRESS);
+    require(
+      _getEOAOwner(msg.sender) == rewardProgramAdmin,
+      "owner of safe is not reward program admin"
+    );
+    require(
+      rewardBalance[rewardProgramID][token] >= amount,
+      "not enough tokens to withdraw"
+    );
+    rewardBalance[rewardProgramID][token] = rewardBalance[rewardProgramID][
+      token
+    ].sub(amount);
+    IERC677(token).transfer(msg.sender, amount);
+    emit RewardTokensExtracted(
+      rewardProgramID,
+      token,
+      amount,
+      rewardProgramAdmin
+    );
+  }
+
+  // lazy implementation of getting eoa owner of safe that has 1 or 2 owners
+  // think this is a use-case to handle during safe manager refactor
+  function _getEOAOwner(address safe) internal returns (address) {
```

Comment:
Yeah I think this definitely is not guaranteed to be right, I agree that the refactor will solve